storage/s3_test.go

```diff
@@ -22,25 +22,36 @@ import (
 )
 
 func getS3() *storage.S3 {
-	config := &aws.Config{ //nolint: exhaustivestruct
-		Credentials: credentials.NewStaticCredentials(
-			os.Getenv("TEST_S3_ACCESS_KEY"),
-			os.Getenv("TEST_S3_SECRET_KEY"),
-			"",
-		),
-		Endpoint:         aws.String("http://localhost:9000"),
-		Region:           aws.String("eu-central-1"),
-		DisableSSL:       aws.Bool(true),
-		S3ForcePathStyle: aws.Bool(true),
-	}
+	// config := &aws.Config{ //nolint: exhaustivestruct
+	// 	Credentials: credentials.NewStaticCredentials(
+	// 		os.Getenv("TEST_S3_ACCESS_KEY"),
+	// 		os.Getenv("TEST_S3_SECRET_KEY"),
+	// 		"",
+	// 	),
+	// 	Endpoint:         aws.String("http://localhost:9000"),
+	// 	Region:           aws.String("eu-central-1"),
+	// 	DisableSSL:       aws.Bool(true),
+	// 	S3ForcePathStyle: aws.Bool(true),
+	// }
 
 	logger := logrus.New()
+	ctx := context.Background()
+	config, err := config.LoadDefaultConfig(ctx,
+		config.WithRegion("eu-central-1"),
+		config.WithCredentialsProvider(
+			credentials.NewStaticCredentialsProvider(
+				os.Getenv("TEST_S3_ACCESS_KEY"),
+				os.Getenv("TEST_S3_SECRET_KEY"),
+				"",
+			),
+		),
+	)
 
-	url := "http://localhost:9000"
-	st, err := storage.NewS3(config, "default", "f215cf48-7458-4596-9aa5-2159fc6a3caf", url, logger)
 	if err != nil {
-		panic(err)
+		logger.Error(err)
 	}
+	url := "http://localhost:9000"
+	st := storage.NewS3(config, "default", "f215cf48-7458-4596-9aa5-2159fc6a3caf", url, true, logger, ctx)
```

Comment:
instead of passing the config and creating the client inside `NewS3` I think it's better to create the client here and pass it directly.